src/plotting.py

```diff
@@ -1485,6 +1485,165 @@ def draw_fuzzy_beachball(problem, po):
         logger.info('Plot already exists! Please use --force to overwrite!')
 
 
+def draw_hudson(problem, po):
+
+    from pyrocko.plot import beachball, hudson
+    from pyrocko import moment_tensor as mtm
+    from numpy import random
+    if problem.config.problem_config.n_sources > 1:
+        raise NotImplementedError(
+            'Hudson plot is not yet implemented for more than one source!')
+
+    if not po.reference:
+        llk_str = po.post_llk
+        stage = Stage(homepath=problem.outfolder)
+
+        stage.load_results(
+            varnames=problem.varnames,
+            model=problem.model, stage_number=po.load_stage,
+            load='trace', chains=[-1])
+
+        n_mts = len(stage.mtrace)
+        m6s = num.empty((n_mts, 6), dtype='float64')
+        for i, varname in enumerate(
+                ['mnn', 'mee', 'mdd', 'mne', 'mnd', 'med']):
+            m6s[:, i] = stage.mtrace.get_values(
+                varname, combine=True, squeeze=True).ravel()
+
+    else:
+        llk_str = 'ref'
+        m6s = num.empty((1, 6), dtype='float64')
+        for i, varname in enumerate(
+                ['mnn', 'mee', 'mdd', 'mne', 'mnd', 'med']):
+            m6s[:, i] = po.reference[varname].ravel()
+
+    m6s_mean = num.mean(m6s, axis=0)
+    m6s_best = m6s[-1, :]
```

Comment:
the solutions are not sorted, this wont be the best, see get_result_point how to find the best